docs/docs/how-to/plugins-and-themes/creating-a-source-plugin.md

```diff
@@ -996,6 +996,98 @@ Here are some things to keep in mind and some "gotchas" depending on how the CMS
 - When you click the "Open Preview" button in the CMS the `manifestId` in the URL should match the `manifestId` that the source plugin creates from that revision.
 - The node manifests get written out in the `public` dir of your gatsby site, so you can check to manifests on your local disk `/public/__node-manifests/<sourcePluginName>/<manifestId>.json` or you can navigate directly to that piece of content `https://<your-domain>/__node-manifests/<sourcePluginName>/<manifestId>`
 
+### Enabling Image CDN Support
+
+Image CDN is a feature on Gatsby Cloud that provides edge network image processing by waiting to perform image processing until the very first user visit to a page. The processed image is then cached for super quick fetching on all subsequent user views. Enabling it will also speed up local development builds and production builds on other deployment platforms because images from your CMS or data source will only be downloaded if they are used in a created Gatsby page.
+
+Learn more [here](https://www.gatsbyjs.com/blog/image-cdn-lightning-fast-image-processing-for-gatsby-cloud/)
+
+## Integration with source plugins
+
+Starting with `gatsby@next`, image CDN is native in Gatsby Core and a polyfill is available to enable image CDN support down to Gatsby v2.
+
+### `createSchemaCustomization` Node API Additions
+
+To add support to a source plugin, you will need to create a new object type that implements the `Node` and `RemoteFile` interfaces.
+
+```js
+schema.buildObjectType({
+  name: `YourImageAssetNodeType`,
+  fields: {
+    // .. your fields
+  },
+  interfaces: [`Node`, `RemoteFile`],
+})
+```
+
+It is also recommended that you add a polyfill to provide support back through Gatsby v2. To do so, you simply wrap the `buildObjectType` call with the `addRemoteFilePolyfillInterface` polyfill like so.
+
+```js
+import { addRemoteFilePolyfillInterface } from "gatsby-plugin-utils/polyfill-remote-file"
+
+addRemoteFilePolyfillInterface(
+  schema.buildObjectType({
+    name: `YourImageAssetNodeType`,
+    fields: {
+      // your fields
+    },
+    interfaces: [`Node`, `RemoteFile`],
+  }),
+  {
+    schema, // schema you get as an argument from createSchemaCustomization
+    actions, // the redux actions of gatsby also part of createSchemaCustomization
+  }
+)
+```
+
+Implementing the `RemoteFile` interface adds the correct fields to your new GraphQL type and adds the necessary resolvers to handle the type. `RemoteFile` holds the following properties:
+
+- `mimeType: String!`
+- `filename: String!`
+- `filesize: Int`
+- `width: Int`
+- `height: Int`
+- `publicUrl: String!`
+- `resize(width: Int, height: Int, fit: enum): String`
+- `gatsbyImage(args): String`
+
+You might notice that width, height, resize and `gatsbyImage` can be null. This is because the `RemoteFile` interface can also handle assets other than images, like PDF’s.
```

Comment:
```suggestion
You might notice that `width`, `height`, `resize`, and `gatsbyImage` can be `null`. This is because the `RemoteFile` interface can also handle assets other than images, like PDF’s.
```

Reply:
Is it standard in Gatsby docs to use the [Oxford comma](https://en.wikipedia.org/wiki/Serial_comma)?